Source/Core/DolphinQt2/GameList/GameListModel.cpp

```diff
@@ -138,6 +138,63 @@ int GameListModel::columnCount(const QModelIndex& parent) const
   return NUM_COLS;
 }
 
+bool GameListModel::ShouldDisplayGameListItem(const QModelIndex& index) {
+  QSharedPointer<GameFile> game = m_games[index.row()];
+
+
+  const bool show_platform = [&game] {
+    switch (game->GetPlatformID())
+    {
+    case DiscIO::Platform::GAMECUBE_DISC:
+      return SConfig::GetInstance().m_ListGC;
+    case DiscIO::Platform::WII_DISC:
+      return SConfig::GetInstance().m_ListWii;
+    case DiscIO::Platform::WII_WAD:
+      return SConfig::GetInstance().m_ListWad;
+    case DiscIO::Platform::ELF_DOL:
+      return SConfig::GetInstance().m_ListElfDol;
+    default:
+      return false;
+    }
+  }();
```

Comment:
(Or if you really like the look of using `return` here, break it up into two functions: `return ShouldDisplayPlatform(DiscIO::Platform) && ShouldDisplayRegion(DiscIO::Country);`)